clients/src/main/java/org/apache/kafka/clients/producer/KafkaProducer.java

```diff
@@ -1115,6 +1115,10 @@ private void ensureValidRecordSize(int size) {
      */
     @Override
     public void flush() {
+        boolean invokedFromCallback = Thread.currentThread() == this.ioThread;
```

Comment:
As this check is used by ```close``` also, could you add a helper method to return boolean?